taichi/backends/device.h

```diff
@@ -389,19 +389,23 @@ class Device {
     AllocUsage usage{AllocUsage::Storage};
   };
 
+  struct AllocParamsLlvmRuntime : AllocParams {
+    bool use_cached{true};
+    JITModule *runtime_jit{nullptr};
+    LLVMRuntime *runtime{nullptr};
+    uint64 *result_buffer{nullptr};
+  };
+
   virtual DeviceAllocation allocate_memory(const AllocParams &params) = 0;
-  virtual DeviceAllocation allocate_memory_runtime(const AllocParams &params,
-                                                   JITModule *runtime_jit,
-                                                   LLVMRuntime *runtime,
-                                                   uint64 *result_buffer) {
+
+  virtual DeviceAllocation allocate_memory_runtime(
+      AllocParamsLlvmRuntime &params) {
     TI_NOT_IMPLEMENTED
   }
+
   virtual void dealloc_memory(DeviceAllocation handle) = 0;
 
-  uint64_t *allocate_llvm_runtime_memory_jit(JITModule *runtime_jit,
-                                             LLVMRuntime *runtime,
-                                             size_t size,
-                                             uint64 *result_buffer);
+  uint64_t *allocate_llvm_runtime_memory_jit(AllocParamsLlvmRuntime &params);
```

Comment:
Can we make `params` const?

Reply:
ehm, this struct has the runtime_jit and result buffer that prevent const. Maybe a better name is `LlvmRuntimeAllocContext`? which can contain a sub struct that is of params const?